### PR TITLE
Addresses #1264 - Hotfix for rendering nil users for discourse posts

### PR DESCRIPTION
**Pull request by spMohanty**

### CODE REVIEW

File: app/views/participants/_tooltip_profile.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<div class="tooltip-inner-top">
+x<div class="tooltip-inner-top">
```
Comment (nemaniarjun) [inline app/views/participants/_tooltip_profile.html.erb]: accidental change?

File: db/schema.rb [hunk 2]
```diff
@@ -783,6 +783,7 @@
     t.float "temporary_variation"
     t.integer "ranking", default: -1, null: false
     t.integer "ranking_change", default: 0, null: false
+    t.boolean "agreed_to_organizers_newsletter", default: true, null: false
```
Comment (nemaniarjun) [inline db/schema.rb]: Unnecessary change. 
Remove this file from the commit

File: app/views/shared/discourse/_topics_list.html.erb [hunk 3]
```diff
@@ -26,7 +26,11 @@
 
       <div class="avatar-group col-2">
         <% discourse_topic['posters'].take(4).each_with_index do |poster, index| %>
-          <%= render partial: 'participants/user_link', locals: { participant: poster['participant'], username: false } %>
+          <% if poster['participant'].present? %>
```
Comment (nemaniarjun) [inline app/views/shared/discourse/_topics_list.html.erb]: This change should be made in the partial as there might be other places where the partial is used

File: app/views/participants/_tooltip_profile.html.erb (hunk 1 above)
Comment (spMohanty) [reply app/views/participants/_tooltip_profile.html.erb]: Yes, an accidental change. Fixed in a subsequent commit.

File: db/schema.rb (hunk 2 above)
Comment (spMohanty) [reply db/schema.rb]: agree, this also got added by mistake.
But this version of `schema.rb` should anyway be reflected in the master branch right ?

File: app/views/shared/discourse/_topics_list.html.erb (hunk 3 above)
Comment (spMohanty) [reply app/views/shared/discourse/_topics_list.html.erb]: Agree. 
I will let @skbly7 deal with the refactoring there, as he set that whole thing up.
At the moment, pushing this to ensure the dev setup is not broken atleast. Others had also faced the same issue.

File: db/schema.rb (hunk 2 above)
Comment (piotrekpasciak) [reply db/schema.rb]: This change actually is good. Master branch is missing, because I incorrectly resolved the conflics. I recommend keeping it in this PR. Otherwise I'll have to readd it in other PR.

File: app/views/shared/discourse/_topics_list.html.erb (hunk 3 above)
Comment (skbly7) [reply app/views/shared/discourse/_topics_list.html.erb]: ACK, no harm and this should anyway never happen in the production environment. 

As we discussed previously, we need to add necessary checks in Discourse integration so it doesn’t active in the dev environment, which still seems to be the case. @piotrekpasciak
